### PR TITLE
Add using should not make innapropriate spell checking fixes.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Tests forthcoming.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Fixes https://github.com/dotnet/roslyn/issues/10911

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by jasonmalinowski** [comment]
:+1:, but it might help to have a comment on each place where you're specifying true/false to explain _why_ that's the particular value chosen.

File: src/Workspaces/CoreTest/UtilityTest/SpellCheckerTests.cs [hunk 1]
```diff
@@ -15,10 +15,14 @@ public class WordSimilarityCheckerTests
...
+            Assert.True(WordSimilarityChecker.AreSimilar("variabledeclaratorsyntax", "variabledeclaratorsyntaxextensions", substringsAreSimilar: true));
 
-            Assert.True(WordSimilarityChecker.AreSimilar("expressionsyntax", "expressionsyntaxextensions"));
-            Assert.True(WordSimilarityChecker.AreSimilar("expressionsyntax", "expressionsyntaxgeneratorvisitor"));
+            Assert.False(WordSimilarityChecker.AreSimilar("expressionsyntax", "expressionsyntaxextensions"));
+            Assert.True(WordSimilarityChecker.AreSimilar("expressionsyntax", "expressionsyntaxextensions", substringsAreSimilar: true));
+
+            Assert.False(WordSimilarityChecker.AreSimilar("expressionsyntax", "expressionsyntaxgeneratorvisitor"));
+            Assert.True(WordSimilarityChecker.AreSimilar("expressionsyntax", "expressionsyntaxgeneratorvisitor", substringsAreSimilar: true));
         }
```
Comment (balajikris) [inline src/Workspaces/CoreTest/UtilityTest/SpellCheckerTests.cs]: these test the `<substring><suffix>` case. do we need tests for substring matches occuring anywhere in the string, say: `<prefix><substring>, <prefix><substring><suffix>`, casing etc?

Comment (balajikris) [inline src/Workspaces/CoreTest/UtilityTest/SpellCheckerTests.cs]: some negative tests where things that aren't similar don;t match, even if the optional parameter is set to `true` ?

**Comment by balajikris** [comment]
:+1:

**Pull request merged**